presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestRemoveNullRowInAggregation.java

```diff
@@ -0,0 +1,134 @@
+                            .addAggregation(p.variable("b"), p.rowExpression("count(a)"))
+                            .source(p.values(p.variable("a")));
+                })).doesNotFire();
+    }
+
+    @Test
+    public void testDoesNotFireOnWithOnlyNullInputFunction()
+    {
+        tester().assertThat(new RemoveNullRowInAggregation(tester.getMetadata().getFunctionAndTypeManager()))
+                .setSystemProperty(REMOVE_NULL_ROW_IN_AGGREGATION, "true")
+                .on(p -> p.aggregation(af -> {
+                    p.variable("a", BIGINT);
+                    p.variable("b", BIGINT);
+                    af.globalGrouping()
+                            .step(AggregationNode.Step.SINGLE)
+                            .addAggregation(p.variable("b"), p.rowExpression("count(*)"))
+                            .source(p.values(p.variable("a")));
+                })).doesNotFire();
+    }
+
+    @Test
+    public void tesDoesNotFireOnWithOneNullInputFunction()
+    {
+        tester().assertThat(new RemoveNullRowInAggregation(tester.getMetadata().getFunctionAndTypeManager()))
+                .setSystemProperty(REMOVE_NULL_ROW_IN_AGGREGATION, "true")
+                .on(p -> p.aggregation(af -> {
+                    p.variable("a", BIGINT);
+                    p.variable("b", BIGINT);
+                    p.variable("c", BIGINT);
+                    af.globalGrouping()
+                            .step(AggregationNode.Step.SINGLE)
+                            .addAggregation(p.variable("c"), p.rowExpression("count(a)"))
+                            .addAggregation(p.variable("b"), p.rowExpression("count(*)"))
+                            .source(p.values(p.variable("a")));
+                })).doesNotFire();
+    }
+
+    @Test
+    public void testRemoveNullRowInAggregationPlan()
+    {
+        ExpectedValueProvider<FunctionCall> aggregationPattern = PlanMatchPattern.functionCall("count", false, ImmutableList.of(anySymbol()));
+        tester().assertThat(new RemoveNullRowInAggregation(tester.getMetadata().getFunctionAndTypeManager()))
+                .setSystemProperty(REMOVE_NULL_ROW_IN_AGGREGATION, "true")
+                .on(p -> p.aggregation(af -> {
+                    p.variable("a", BIGINT);
+                    p.variable("b", BIGINT);
+                    af.globalGrouping()
+                            .step(AggregationNode.Step.SINGLE)
+                            .addAggregation(p.variable("a"), p.rowExpression("count(b)"))
+                            .source(p.values(p.variable("b")));
+                })).matches(aggregation(
+                        globalAggregation(),
+                        ImmutableMap.of(Optional.empty(), aggregationPattern),
+                        ImmutableMap.of(),
+                        Optional.empty(),
+                        SINGLE,
+                        node(FilterNode.class, values("b"))));
+    }
+
+    @Test
+    public void testRemoveNullRowInAggregationResult()
```

Comment:
Check "AbstractTestQueries" class, this test should belong there

Reply:
fixed